src/parsec.rs

```diff
@@ -31,24 +33,58 @@ pub type Parsec = inner::Parsec<chain::NetworkEvent, FullId>;
 pub type Request = inner::Request<chain::NetworkEvent, id::PublicId>;
 pub type Response = inner::Response<chain::NetworkEvent, id::PublicId>;
 
+// TODO: These two needs to be determined from better data.
+// For now: 640kb ought to be enough for anybody!
+const PARSEC_SIZE_LIMIT: u64 = 640_000;
+// The maximum number of handled requests/responses
+const PARSEC_EVENT_LIMIT: u64 = 1000;
+
+// Keep track of size in case we need to prune.
+#[derive(Default, Debug, PartialEq, Eq)]
+struct ParsecSizeCounter {
+    event_count: u64,
+    size_count: u64,
+}
+
+impl ParsecSizeCounter {
+    fn new() -> Self {
+        Default::default()
+    }
+
+    fn increase_size(&mut self, serialised_size: u64) {
+        self.event_count += 1;
+        self.size_count += serialised_size;
+    }
+
+    fn needs_pruning(&self) -> bool {
+        (self.size_count > PARSEC_SIZE_LIMIT) || (self.event_count > PARSEC_EVENT_LIMIT)
```

Comment:
Might it be good to have PARSEC_SIZE_LIMIT be the default initialization for a member here, so you can use different limit in your unit test (also maybe it could be dynamic in the future), but also that mean the the test that change logic do not need to change when limit change.

not sure about that.

Also it might be good in tests to reduce that limit a lot so pruning happen more regularly (and also we got a bunch of nodes running so we have less memory for each).

Reply:
similiar idea as in my comment of https://github.com/maidsafe/routing/pull/1786/files#r324227079